### PR TITLE
Fix issues with python packaging and add more python repo for testing

### PR DESCRIPTION
**Pull request by arroyc**

### CODE REVIEW

**Review by kichalla — APPROVED** [review]

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -45,9 +45,6 @@ fi
 	then
 		echo "Running python setup.py install..."
 		$python setup.py install --user| ts $TS_FMT
-		cd *.egg-info
-		python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r requires.txt | ts $TS_FMT
```
Comment (arroyc) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:49]: we dont need this to build when we only have setup.py and no req.txt

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 2]
```diff
@@ -84,8 +81,6 @@ fi
 
 		echo "Running python setup.py install..."
 		$python setup.py install --user| ts $TS_FMT
-		cd *.egg-info
-		$python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r requires.txt | ts $TS_FMT
```
Comment (arroyc) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:88]: same

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 3]
```diff
@@ -32,6 +32,16 @@ public PythonPackageTest(ITestOutputHelper output) : base(output)
                 "git://github.com/boto/botocore.git" },
             new object[] { "pyasn1", "0.4.8",
                 "git://github.com/etingof/pyasn1.git", "v0.4.8"},
+            new object[] { "configparser", "5.0.0",
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:35]: @cormacpayne take a look at this tests

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 4]
```diff
@@ -113,6 +122,84 @@ public PythonPackageTest(ITestOutputHelper output) : base(output)
                 $"Size difference is too big. Oryx build: {oryxTarSize}, Actual PyPi: {pypiTarSize}");
         }
 
+        [Theory]
+        [InlineData("numpy", "1.19.2", "git://github.com/numpy/numpy.git", "v1.19.2")]
+        public void CanBuildPython3NumpyPackage(
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:127]: @cormacpayne  this is the numpy test

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 5]
```diff
@@ -113,6 +122,84 @@ public PythonPackageTest(ITestOutputHelper output) : base(output)
...
+
+            var (oryxTarList, oryxTarSize) = ParseTarList(tarLists[1]);
+            var (pypiTarList, pypiTarSize) = ParseTarList(tarLists[2]);
+
+            var unContained = pypiTarList.Where(x => !oryxTarList.Contains(x));
+            Assert.Equal(pypiTarList, oryxTarList);
+
+            // Assert tar file sizes
+            var tarSizeDiff = Math.Abs(pypiTarSize - oryxTarSize);
+            Assert.True(tarSizeDiff <= pypiTarSize * 0.1, // Accepting differences of less than 10% of the official artifact size
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:198]: we check tar file size

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 6]
```diff
@@ -113,6 +122,84 @@ public PythonPackageTest(ITestOutputHelper output) : base(output)
...
+            var result = _dockerCli.Run(image, "/bin/bash", new[] { "-c", script });
+
+            // Assert contained file names
+            var tarLists = result.StdOut.Split(tarListMarker);
+
+            var (oryxTarList, oryxTarSize) = ParseTarList(tarLists[1]);
+            var (pypiTarList, pypiTarSize) = ParseTarList(tarLists[2]);
+
+            var unContained = pypiTarList.Where(x => !oryxTarList.Contains(x));
+            Assert.Equal(pypiTarList, oryxTarList);
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:194]: we compare the tar list as well

File: images/build/Dockerfiles/ltsVersions.Dockerfile [hunk 7]
```diff
@@ -28,7 +28,9 @@ RUN LANG="C.UTF-8" \
         python3-pip \
     && rm -rf /var/lib/apt/lists/* \
     && pip install pip --upgrade \
+    && pip install --upgrade cython \
     && pip3 install pip --upgrade \
+    && pip3 install --upgrade cython \
```
Comment (arroyc) [inline images/build/Dockerfiles/ltsVersions.Dockerfile:33]: we need cython in build image to be able to build native python packages/modules

File: tests/Oryx.BuildImage.Tests/Settings.cs [hunk 8]
```diff
@@ -10,7 +10,9 @@ namespace Microsoft.Oryx.BuildImage.Tests
     internal static class Settings
     {
         public const string BuildImageName = "oryxtests/build:latest";
+        public const string BuildImageWithRootAccess = "oryx/build:latest";
         public const string LtsVersionsBuildImageName = "oryxtests/build:lts-versions";
+        public const string LtsVersionsBuildImageWithRootAccess = "oryx/build:lts-versions";
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/Settings.cs:15]: added to ease local troubleshooting

**Review by qianz2 — APPROVED** [review]

**Pull request merged**